cpp/dolfinx/graph/AdjacencyList.h

```diff
@@ -191,8 +191,14 @@ class AdjacencyList
   template <typename X>
   decltype(auto) as_type() const
   {
+// Workaround for Intel compler bug, see
+// https://community.intel.com/t5/Intel-C-Compiler/quot-if-constexpr-quot-and-quot-missing-return-statement-quot-in/td-p/1154551
+#ifdef __INTEL_COMPILER
```

Comment:
Indented preprocessor directives (should) be OK?

Reply:
This is a style thing, ignore me, it's fine.

Reply:
I just let `clang-format` do its thing.